### DISCUSSION TITLE
Target temperature change behavior

### DISCUSSION THREAD

**Discussion by sleveille4** [jmcollin78/versatile_thermostat#578] [General]
Hi,
not sure yet if this is a bug or not, hence why i post in discussion.

i use VT for both water radiator with trvs & electric ones, trv in "over valve"  and radiator in "over switch"

i noticed that the behavior is not the same when changing target temperature.

"over switch" seems to re-start a compute + a new cycle every time the target temp is changed, doesn't matter how quick those are

"over valve" does not.

here are 2 examples scenarios:

"over switch" VT
1- start the climate => it starts in ECO mode (say 15°c)
2- it computes it power % and start it's cycle (5m in my case)
3- switch to comfort mode (18°c) +- 10 seconds later
4- it computes again, update the power % and (i assume) restart a new 5m cycle

same scenario with an "over valve" VT

1- start the climate => it starts in ECO mode (say 15°c)
2- it computes it power % and start it's cycle (5m in my case)
3- switch to comfort mode (18°c) +- 10 seconds later
4- nothing happens, and the change will only be applied after the ongoing cycle is complete

i don't recall VT behaving this way a few month ago, is this by design ? (not a BIG deal, but still might hide a bug under it)

here is a video showing the difference:

https://github.com/user-attachments/assets/969ccb32-4563-4077-b50e-1eda6415c018

**Comment by jmcollin78** [comment]
Hello,

Interesting post. The behavior should be the same : when you change a preset, calculation is done immediatly and cycle is updated.

But in case of TRV, there is some configuration parameters drawing the battery too fast. Those paramters are thresholds that makes the update not mandatory send to the TRV:

![Capture d’écran 2024-10-26 à 11 53 55](https://github.com/user-attachments/assets/f1d5382a-eb58-4afd-b22e-a783b5deb18c)


I guess you are in that case.

**Comment by sleveille4** [comment]
hmmm possible.
i'll check more with those settings as they should be consistent for both unit type (i never changed them), so i would expect either both to update, or not update, but currently they behave differently....
